### ISSUE TITLE
Error 259 during WinRM Provisioning

### ISSUE THREAD

**Issue by mefellows** [packer-community/packer-windows-plugins#33]
I've had a number of these occur over the past couple of days when using the Elevated Powershell runner:

```
==> buildagent: Inspecting the source AMI...
==> buildagent: Creating temporary security group for this instance...
==> buildagent: Authorizing WinRM access on the temporary security group...
==> buildagent: Launching a source AWS instance...
    buildagent: Instance ID: i-9209d54e
==> buildagent: Waiting for instance (i-9209d54e) to become ready...
==> buildagent: Waiting for WinRM to become available...
==> buildagent: Connected to WinRM!
==> buildagent: Provisioning with Powershell...
==> buildagent: Provisioning with shell script: /var/folders/kf/4sgp93ys2t3_9t3yyzr07c3r0000gn/T/packer-powershell-provisioner400494783
==> buildagent: Provisioning with Powershell...
==> buildagent: Provisioning with shell script: ./scripts/provision-agent.ps1
    buildagent: !!ATTENTION!!
    buildagent: The next version of Chocolatey (v0.9.9) will require -y to perform
    buildagent: behaviors that change state without prompting for confirmation. Start
    buildagent: using it now in your automated scripts.
    buildagent:
    buildagent: For details on the all new Chocolatey, visit http://bit.ly/new_choco
==> buildagent: Terminating the source AWS instance...
==> buildagent: Deleting temporary security group...
Build 'buildagent' errored: Script exited with non-zero exit status: 259
```

As it turns out, this issue has also been experienced in [Vagrant](https://github.com/mitchellh/vagrant/issues/5362) and has recently been [fixed](https://github.com/mitchellh/vagrant/pull/5469].

This potentially raises another issue with some powershell scripts which may return error codes that aren't really error codes (Chocolatey explicitly deals with this situation by asking for an array of valid exit codes - for example 0,3010 for installing VS). For now, this should be the responsibility of the powershell script and perhaps we ought to update the docu to spell out the obvious - non-0 exit codes result in build failures.

Anyway, I'll implement the same change locally and if these issues disappear i'll push later on today.

**Comment by mefellows** [comment]
Haven't experienced this issue since c3fcd2b and fixes for #37 and #36

**Comment by colinbowern** [comment]
Using the pre-release bits and I'm getting this sporadically for a PSH script that installs VMware Tools:

    ==> vmware-windows-iso: Downloading or copying ISO
        vmware-windows-iso: Downloading or copying: file://C:\Users\colinb\Projects\BaseBoxes/C:/Users/colinb/Projects/BaseBoxes/Images/WindowsServer-2012R2.1-amd64.iso
    ==> vmware-windows-iso: Creating floppy disk...
        vmware-windows-iso: Copying: Keys/CertificateAuthority.crt
        vmware-windows-iso: Copying: Keys/Work/BASEWS2012R21.pfx
        vmware-windows-iso: Copying: Scripts/Enable-WinRM.ps1
        vmware-windows-iso: Copying: Templates/WindowsServer-2012R2.1-Standard/Autounattend.xml
        vmware-windows-iso: Copying: Templates/WindowsServer-2012R2.1-Standard/Sysprep.xml
    ==> vmware-windows-iso: Creating virtual machine disk
    ==> vmware-windows-iso: Building and writing VMX file
    ==> vmware-windows-iso: Starting virtual machine...
    ==> vmware-windows-iso: Waiting 10s for boot...
    ==> vmware-windows-iso: Connecting to VM via VNC
    ==> vmware-windows-iso: Typing the boot command over VNC...
    ==> vmware-windows-iso: Waiting for WinRM to become available...
    ==> vmware-windows-iso: Connected to WinRM!
    ==> vmware-windows-iso: Uploading the 'windows' VMware Tools
    ==> vmware-windows-iso: Provisioning with Powershell...
    ==> vmware-windows-iso: Provisioning with shell script: Scripts/Install-VMwareTools.ps1
        vmware-windows-iso: Transcript started, output file is C:\Users\vagrant\AppData\Local\Temp\script.log
        vmware-windows-iso: #< CLIXML
        vmware-windows-iso: Transcript stopped, output file is C:\Users\vagrant\AppData\Local\Temp\script.log
        vmware-windows-iso:
        vmware-windows-iso: <Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04"><Obj S="progress" RefId="0"><TN RefId="0"><T>System.Management.Automation.PSCustomObject</T><T>System.Object</T></TN><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Mount-DiskImage -ImagePath $Env:Temp\VMwareTools.iso</AV><AI>59606033</AI><Nil /><PI>-1</PI><PC>0</PC><T>Processing</T><SR>-1</SR><SD>0/2+ completed</SD></PR></MS></Obj><Obj S="progress" RefId="1"><TNRef RefId="0" /><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Mount-DiskImage -ImagePath $Env:Temp\VMwareTools.iso</AV><AI>59606033</AI><Nil /><PI>-1</PI><PC>100</PC><T>Completed</T><SR>0</SR><SD>2/2 completed</SD></PR></MS></Obj><Obj S="progress" RefId="2"><TNRef RefId="0" /><MS><I64 N="SourceId">2</I64><PR N="Record"><AV>Get-DiskImage -ImagePath $Env:Temp\VMwareTools.iso | Get-Volume</AV><AI>496342505</AI><Nil /><PI>-1</PI><PC>100</PC><T>Completed</T><SR>0</SR><SD>1/1 completed</SD></PR></MS></Obj><Obj S="progress" RefId="3"><TNRef RefId="0" /><MS><I64 N="SourceId">3</I64><PR N="Record"><AV>Get-Volume</AV><AI>607711163</AI><Nil /><PI>-1</PI><PC>100</PC><T>Completed</T><SR>0</SR><SD>1/1 completed</SD></PR></MS></Obj><Obj S="progress" RefId="4"><TNRef RefId="0" /><MS><I64 N="SourceId">4</I64><PR N="Record"><AV>Dismount-DiskImage -ImagePath $Env:Temp\VMwareTools.iso</AV><AI>1144664691</AI><Nil /><PI>-1</PI><PC>100</PC><T>Completed</T><SR>0</SR><SD>2/2 completed</SD></PR></MS></Obj></Objs>
    ==> vmware-windows-iso: Stopping virtual machine...
    ==> vmware-windows-iso: Deleting output directory...
    Build 'vmware-windows-iso' errored: Script exited with non-zero exit status: 259. Allowed exit codes are: [%!s(int=0)]

    ==> Some builds didn't complete successfully and had errors:
    --> vmware-windows-iso: Script exited with non-zero exit status: 259. Allowed exit codes are: [%!s(int=0)]

    ==> Builds finished but no artifacts were created.

With Packer destroying the artifacts it is difficult to inspect what happened. Packer really needs an option to preserve failed output.
